app.rb

```diff
@@ -111,4 +114,36 @@ def list_rentals_by_id
       puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}" if rental.person.id == id
     end
   end
+
+  def preserve_data
+    preserve_books if @books.length.positive?
+    preserve_persons if @persons.length.positive?
+    preserve_rentals if @rentals.length.positive?
+  end
```

Comment:
- [ ] Great job so far team. You have done an amazing job so far and your app is able to preserve data as expected :clap: 
- [ ] However, I deleted all the current `JSON` files in your project to test if the part of the project requirement that mentioned that you need to remember to handle the case when one or more of the files are missing and ensure that the app works and shows no errors 👍 
- [ ] The app did start as expected. However, when some inputs were made and even in an attempt to create new books, people, and rentals, the app threw some errors and stopped working. 
- [ ] I also noticed that when I tried to exit the app, the `JSON` files weren't recreated in the `files` where they were previously to store the new data since new data wasn't even created. As a result, data wasn't persisted in your app as expected so kindly fix this and ensure that your app works as expected even if any of the JSON files are missing 😉 
- [ ] If you look at the video, you'll notice that all the features of your app stopped working as intended after the files were deleted even the option to exit the app failed to work as expected. So kindly work on this so your app works as expected :100: 
- Video Reference
_**I took this video after deleting all the JSON files from the project 👍**_

[Screencast from 06-27-2023 03:57:30 PM.webm](https://github.com/dev89s/school-library/assets/102699958/9f62191e-d88b-4587-b63e-83b59f5bb520)

- Project Requirement
![image](https://github.com/dev89s/school-library/assets/102699958/dc3da349-ee7c-4ca1-853d-7f0f8f610931)